steps to use the python commit script

choldgraf opened the pull request:
This updates the deploy steps to use the `list_new_commits.py` script. I hope this will reduce the number of steps needed to make a deploy, and also standardize the PR names that we've got. Lemme know what folks think!

willingc commented on docs/source/deployment/how.md:
```diff
@@ -24,35 +24,41 @@ which we step through below.
-   be published and listed on GitHub. Go to the [BinderHub Helm Chart](https://jupyterhub.github.io/helm-chart/#development-releases-binderhub)
-   page and grab the hash for the latest published version (at the top).
+3. Wait for the build for your PR merge to pass (it will say "#NNN passed").
+   If it does, then:
```
If it does, continue to Step 4? If it doesn't ???

jzf2101 replied:
Meta question here- is this doc addressed to people who have merge rights already? Because the if it doesn't depends on the audience AFAIK

jzf2101 replied:
Meta question- who is the audience here? If the person is a maintainer vs contributor the answer will be difference if the build doesn't pass

willingc commented on docs/source/deployment/how.md:
```diff
@@ -24,35 +24,41 @@ which we step through below.
+       ---------------------
 
-6. In your fork of the [mybinder.org-deploy](https://github.com/jupyterhub/mybinder.org-deploy)
+       BinderHub: https://github.com/jupyterhub/binderhub/compare/<OLD-HASH>...<NEW-HASH>
```
Should we add JupyterHub as well? If it hasn't changed, no big deal. If it has, the information gets captured aas well.

choldgraf replied:
yeah we definitely could to the `.py` script. Do you know where in BinderHub is the value for which jupyterhub we depend on?

willingc replied:
https://github.com/jupyterhub/binderhub/blob/master/helm-chart/binderhub/requirements.yaml Perhaps just print JupyterHub as an informational message. It's cool to leave it off too.

jzf2101 commented on docs/source/deployment/how.md:
```diff
@@ -24,35 +24,41 @@ which we step through below.
+
+   Since you are updating BinderHub, copy the text in `<NEW-HASH>` for the
+   line that refers to BinderHub. This is the name of the new BinderHub image.
+   We'll now update the config to refer to this image.
```
Does we refer to the maintainers here?

willingc replied:
These docs render to the mybinder Site Reliability Guide. We probably can have a blanket statement that this document is for maintainers of mybinder.org

jzf2101 commented on docs/source/deployment/how.md:
```diff
@@ -24,35 +24,41 @@ which we step through below.
+6. Toward the end of the file, you will see lines similar to:
 
       - name: binderhub
         version: 0.1.0-9692255
```
Shouldn't this be `<OLD-HASH>` then?

jzf2101 commented on docs/source/deployment/how.md:
```diff
@@ -24,35 +24,41 @@ which we step through below.
+   above case, we'll replace it with the new hash.
 
       - name: binderhub
         version: 0.1.0-fbf6e5a
```
Again this would likely be `<NEW-HASH>` by the same logic?

jzf2101 commented on docs/source/deployment/how.md:
```diff
@@ -68,27 +74,35 @@ The following lines describe how to point mybinder.org to the new repo2docker im
-   `mybinder/values.yaml`.
-7. Somewhere in the file you will see `repo2dockerImage`, it will look like
+   And click on the number for the latest build on "Master".
+3. Wait for the build to pass (it will say "#NNN passed"). If it does, then:
```
By your same logic @willingc we should have something if this fails?

willingc replied:
+1

jzf2101 commented on docs/source/deployment/how.md:
```diff
@@ -68,27 +74,35 @@ The following lines describe how to point mybinder.org to the new repo2docker im
+       repo2dockerImage: jupyter/repo2docker:<NEW-HASH>
 
 8. Merge this change to `mybinder/values.yaml` into the mybinder.org-deploy
    repository following the steps in the [Deploying a change](#deploying-a-change) section
```
More broadly perhaps we can consolidate a lot of these steps either here or in "Deploying a Change" since the steps are rather similar.

choldgraf commented:
Hey all - picking up my old PRs now that I'm married and don't have to plan 2 weddings anymore :-)

I updated this PR to incorporate the JupyterHub commits as well, so we can compare all three. I also incorporated most of the feedback that @willingc and @jzf2101 gave (thanks!).

Can we leave the larger organizational comments (e.g. refactoring the whole page so there's less repeated content) to another PR? :-)

betatim commented:
Can we keep the (repo2docker) instructions so that they can be used without installing a script?

I've been asking people who contribute to r2d if they also want to deploy the change and point them at the SRE guide for instructions. So far people have been happy with them and the success rate of people agreeing to do it and doing it is high. I don't know if this is because you can do all the steps from a browser but it feels like having to fork, clone, run a script, go to GitHub has more friction :-/ Or it would lead to people being more invested because they have made some effort setting stuff up :-/

Is there an easy compromise?

choldgraf commented:
I don't think this script is really "required", you can still look up the commit from the history, it's just meant to be a helper to make it easier for people who want to make comparison URLs. If people don't think this would make it easier, then I agree we shouldn't have the script up there...

betatim commented:
Ah ok. From a quick look at the diff this morning it looked like we were removing the steps for looking it up by hand. If we keep the "by hand" instructions my concerns are null and void.

The script definitely makes my life easier (so I want this script) was more thinking about new contributors and what a reasonable number of steps is to ask them do.

choldgraf commented:
@betatim I mean, we could always build a Binder that lets people run the script :-)  all that information is in the public anyway...

choldgraf commented:
mergable from my end
